## 9.4 Probability to odds

##### a. If the probability of an event occurring is 0.8, what are the odds?

The odds are $\frac{p}{1-p} = \frac{.8}{.2} = 4$ (or 4 to 1)

##### b. If the probability of an event occurring is 0.25, what are the odds?

Similarly, the odds are $\frac{p}{1-p} = \frac{.25}{.75} = .333$ (or 1 to 3)

##### c. If the probability of an event occurring is 0.6, what are the odds?

The odds are $\frac{p}{1-p} = \frac{.6}{.4} = 1.5$ (or 3 to 2)

## 9.6 Odds to Probability

##### a. If the odds of an event occurring are 1:3, what is the probability? 
If $P = \frac{\text{successes}}{\text{successes + failures}}$, then the probability of 1:3 odds is $\frac{1}{1+3}= \frac{1}{4} = 0.25$

##### b. If the odds of an event occurring are 5:2, what is the probability? 
Similarly, $P = \frac{5}{7} = .714$

##### c. If the odds of an event occurring are 1:9, what is the probability? 
$P = \frac{1}{10} = .1$, or 10%

## 9.21 Metastizing Cancer: Odds and probability

##### a. Use this model to estimate the odds of metastasis, $\frac{\pi}{1 - \pi}$ if a patient's tumor size is 6cm.

$-2.086 + .5116(6) = -2.086 + 3.0696 = .9836 = exp(.9836) = 2.67$. So the odds are roughly 2.7 to 1

##### b. Use the model to predict the probability of metastasis if a patient's tumor is 6cm

$P = \frac{2.67}{3.67} = .728$ or roughly 73%.

##### c. How much do the estimated odds change if the tumor size changes from 6 to 7 cm?

At 7cm, the logistic model gives us $exp(-2.086 + .5116 * 7) = 4.459$, or roughly 4.6 to 1 odds. The absolute change in odds is 4.4459 - 2.673 = 1.786.

## 9.23 Estimating X from probability

##### Consider the fitted logit model from exercise 9.21 How large does a tumor need to be for the estimated probability of metastasis to be at least .8?

We want the odds to reach at least 4:1 (since $\frac{.8}{1.8} = 4$):
$-2.086 + .5117S = ln(4) = 1.3863$

Solving for S,
$S = \frac{ln(4) + 2.086}{.5117} \approx 6.79$

So the tumor must be about 6.8 cm or larger for the model to predict a probability of metastasis of at least .8. 



## 9.22 Dementia: Odds and Probability

##### a. Use this model to estimate the odds of Alzheimer's disease if a patient's MMSE is -4

$L = -.742 - 2.94(-4) = 11.018 = exp(11.018) = 6.1 x 10^4$, or roughly 61000 to 1

##### b. Use this model to estimate the probability of Alzheimer's disease if a patient's MMSE is -4.

$\frac{6.1 * 10^4}{1 + 6.1 * 10^4} = .99998$, or 99.998%

##### c. How much do the estimated odds change if the MMS# changes from -4 to -3?

Odds at -4 = 6.1
Odds at -3 = $exp(-.742 - 2.94 * -3) = 3.23 * 10^3$

Ratio: $\frac{odds at -3}{odds at -4} = \frac{3.23 * 10^3}{6.1 * 10^4} = .053$

So moving from -4 to -3 reduces the odds to about 5.3 of their previous value.

## 9.24 Estimating X from probability

##### Consider the fitted logit model from exercise 9.22. What value of MMSE is necessary to have a probability of Alzheimer's of at least .75?


We want the odds to reach at least 3:1 (since $\frac{.75}{.25} = 3$):
$ln(3) \approx 1.0986$

$-.742 - 2.94m = 1.0986 => m = -\frac{1.8406}{2.94} \approx -.626$

We'd need an MMSE of approximately -.63.

## Problem 1 (basedon 9.36 and 9.38)

##### a. Create the simple logistic models in problem 9.36

In [2]:
election <- read.csv("./data/Election16.csv")

# TrumpWin ~ Income
mod_income <- glm(TrumpWin ~ Income, data = election, family = binomial)
summary(mod_income)

# rumpWin ~ HS
mod_HS <- glm(TrumpWin ~ HS, data = election, family = binomial)
summary(mod_HS)

# TrumpWin ~ BA
mod_BA <- glm(TrumpWin ~ BA, data = election, family = binomial)
summary(mod_BA)

# TrumpWin ~ Dem.Rep
mod_DemRep <- glm(TrumpWin ~ Dem.Rep, data = election, family = binomial)
summary(mod_DemRep)


Call:
glm(formula = TrumpWin ~ Income, family = binomial, data = election)

Coefficients:
              Estimate Std. Error z value Pr(>|z|)    
(Intercept)  1.118e+01  3.076e+00   3.635 0.000277 ***
Income      -1.967e-04  5.582e-05  -3.523 0.000426 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 67.301  on 49  degrees of freedom
Residual deviance: 45.923  on 48  degrees of freedom
AIC: 49.923

Number of Fisher Scoring iterations: 5



Call:
glm(formula = TrumpWin ~ HS, family = binomial, data = election)

Coefficients:
            Estimate Std. Error z value Pr(>|z|)
(Intercept)  9.06905    8.64223   1.049    0.294
HS          -0.09809    0.09768  -1.004    0.315

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 67.301  on 49  degrees of freedom
Residual deviance: 66.262  on 48  degrees of freedom
AIC: 70.262

Number of Fisher Scoring iterations: 4



Call:
glm(formula = TrumpWin ~ BA, family = binomial, data = election)

Coefficients:
            Estimate Std. Error z value Pr(>|z|)    
(Intercept)  17.9973     5.1098   3.522 0.000428 ***
BA           -0.5985     0.1735  -3.449 0.000562 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 67.301  on 49  degrees of freedom
Residual deviance: 34.433  on 48  degrees of freedom
AIC: 38.433

Number of Fisher Scoring iterations: 6



Call:
glm(formula = TrumpWin ~ Dem.Rep, family = binomial, data = election)

Coefficients:
            Estimate Std. Error z value Pr(>|z|)    
(Intercept)  0.32273    0.43466   0.742 0.457788    
Dem.Rep     -0.25034    0.07447  -3.361 0.000775 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 67.301  on 49  degrees of freedom
Residual deviance: 33.596  on 48  degrees of freedom
AIC: 37.596

Number of Fisher Scoring iterations: 6


Based on the AIC across the models and the deviance, Dem.Rep seems to be the best predictor, while HS is the weakest.

##### b. Read and do the instructions in 9.38, and answer the questions in parts a, b, and c.

##### a. Run the logistic regression using IncomeTh as the predictor of TrupmWin. How does the fitted prediction equation change? How (if at all) does the predicted probability of Trump winning a state change?

In [3]:
election <- read.csv("./data/Election16.csv")
election$IncomeTh <- election$Income / 1000
mod_IncomeTh <- glm(TrumpWin ~ IncomeTh, data = election, family = binomial)
summary(mod_IncomeTh)


Call:
glm(formula = TrumpWin ~ IncomeTh, family = binomial, data = election)

Coefficients:
            Estimate Std. Error z value Pr(>|z|)    
(Intercept) 11.18186    3.07576   3.635 0.000277 ***
IncomeTh    -0.19668    0.05582  -3.523 0.000426 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 67.301  on 49  degrees of freedom
Residual deviance: 45.923  on 48  degrees of freedom
AIC: 49.923

Number of Fisher Scoring iterations: 5


Nothing much has changed - only the numeric value of the coefficient reflects the new "per-thousand" unit.

##### b. Use the estimated slope from the regression to compute an estimated odds ratio, and write a sentence that interprets this value in the context of this problem.

The fitted slope for IncomeTh is -.19668, with a corresponding odds-ratio of exp(-.19668) = .82. This means that for every additional $1000 in per capita income, the estimated odds of a Trump win decrease by about .82.

##### c. Find a 95% confidence interval for the odds ratio in (b).

In [5]:
ci_beta <- confint(mod_IncomeTh)
ci_OR <- exp(ci_beta["IncomeTh", ])
ci_OR

Waiting for profiling to be done...



2.5 %    97.5 % 
0.7232693 0.9037885

The 95% CI for the odds ratio is (0.74, 0.91). This can be interpreted as being 95% confident that each additional $1000 in per-capita income multiplies the odds of a Trump win by between .736 and .91

## Problem 2 (based on problem 9.41)

##### a. Answer parts a and c of problem 9.41

##### a. Fit a logistic model using Resp as the response variable and Age as the predictor variable. Interpret the results and state whether the relationship is statistically significant.

In [6]:
leukemia <- read.csv("./data/Leukemia.csv")
mod_age <- glm(Resp ~ Age, data   = leukemia, family = binomial)
summary(mod_age)


Call:
glm(formula = Resp ~ Age, family = binomial, data = leukemia)

Coefficients:
            Estimate Std. Error z value Pr(>|z|)  
(Intercept)  2.19678    1.00548   2.185   0.0289 *
Age         -0.04676    0.01952  -2.395   0.0166 *
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 70.524  on 50  degrees of freedom
Residual deviance: 64.004  on 49  degrees of freedom
AIC: 68.004

Number of Fisher Scoring iterations: 4


Based on the results, we would say that age is a statistically significant predictor of response, and the drop in deviance (roughly 6.52 on 1df) indicates that adding Age improves fit compared to the intercept only mode.

##### c. Redo parts (a) but using Temp as the single predictor variable

In [7]:
mod_temp <- glm(Resp ~ Temp, data   = leukemia, family = binomial)
summary(mod_temp)


Call:
glm(formula = Resp ~ Temp, family = binomial, data = leukemia)

Coefficients:
            Estimate Std. Error z value Pr(>|z|)  
(Intercept) 38.55067   21.23644   1.815   0.0695 .
Temp        -0.03884    0.02135  -1.819   0.0689 .
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 70.524  on 50  degrees of freedom
Residual deviance: 66.766  on 49  degrees of freedom
AIC: 70.766

Number of Fisher Scoring iterations: 4


We do not get a statistically significant effect from Temp, and the evidence is too weak to conclude that Temp is a significant predictor of Resp in this dataset.

##### b. Also fit a logistic model for Resp using Smear, and interpret the results and discuss whether the relationship is statistically significant as you did for the other variables.



In [8]:
mod_smear <- glm(Resp ~ Smear, data   = leukemia, family = binomial)
summary(mod_smear)


Call:
glm(formula = Resp ~ Smear, family = binomial, data = leukemia)

Coefficients:
            Estimate Std. Error z value Pr(>|z|)
(Intercept) -1.67373    1.07441  -1.558    0.119
Smear        0.02348    0.01552   1.512    0.130

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 70.524  on 50  degrees of freedom
Residual deviance: 68.117  on 49  degrees of freedom
AIC: 72.117

Number of Fisher Scoring iterations: 4


Again, we do not get a statisically significant effect for the variable Smear.

##### c. For each of your three models, also create a 95% confidence interval for the odds ratio for the slope, using the idea on page 438 and taking exponetials, and explain how this confidence interval does or does not agree with your previous discussion of statistical significance.

In [9]:
ci_age_log <- confint(mod_age   )["Age",   ]
ci_temp_log <- confint(mod_temp  )["Temp",  ]
ci_smear_log <- confint(mod_smear)["Smear", ]

ci_or_age <- exp(ci_age_log)
ci_or_temp <- exp(ci_temp_log)
ci_or_smear <- exp(ci_smear_log)

ci_or_age
ci_or_temp
ci_or_smear

Waiting for profiling to be done...

Waiting for profiling to be done...

Waiting for profiling to be done...



2.5 %    97.5 % 
0.9157153 0.9895862

2.5 %    97.5 % 
0.9186037 1.0004083

2.5 %    97.5 % 
0.9939632 1.0571967

The entire 95% CI for the odds-ratio for age lies below 1, so a one-year increase in age is estimated to decrease the odds of response by between roughly 1-8%, not including 1, which agrees with our earlier finding that Age is a significant predictor.

For both Temp and Smear, the CI crosses 1 (only just for Temp), so we cannot rule out no effect, which matches the non-significant p-values we saw for, and also aligns with the findings that Temp appears to have more of an effect than Smear, although again, with no statistical significant at the 5% threshold.